### ISSUE TITLE
Non-existent loot being counted in "newer" heists.

### ISSUE THREAD

**Issue by Malidictus** [Kamikaze94/WolfHUD#654]
Ever since Reservoir Dogs, WolfHUD has been counting loot on maps where I'm pretty sure it doesn't exist. Here are the heists and my reasoning for saying this:

Reservoir Dogs Store: The game gave me the achievement for securing all loot with 1 bag unaccounted for, according to WolfHUD. We searched high and low and could not locate another lootable item. Additionally, it seems like loot in the back of the store and in the vault counts before players can access it, which is not the case on similar heists like Hotline Miami Day 1.

Breakin' Feds: The potential loot counter for the heist is always +1 to the number of bags available for pickup. The Kotaru box doesn't count until Garrett's safe is open, but the evidence room loot does. Except the evidence room doesn't hold enough bags to account for the loot display.

Henry's Rock: The loot counter shows 5 potential loot items when only one exists - the Artefact in Archeology. I don't know what the other 4 are, but I suspect they may be the painting props in Archeology. Additionally, the number of openable crates listed seem +1 to the number of actual openable crates on the map. We've searched everywhere and weren't able to find that last crate to open.

**Comment by MEXAHOTABOP** [comment]
try write issue using this https://github.com/Kamikaze94/WolfHUD/blob/master/.github/ISSUE_TEMPLATE.md

**Comment by Malidictus** [comment]
Please answer the following questions for yourself before submitting an issue.  
- [*] I am running the latest version
- [*] I made sure this issue doesn't persist without this mod
- [*] I checked to make sure that this issue has not already been filed

# Expected Behavior

The HudPanel potential custom loot counter should count only loot and crates which actually exist on the map and can be interacted with.

# Current Behavior

The HudPanel potential custom loot counter counts loot and crates which don't exist on the map or can't be interacted with.

# Failure Information (for bugs)

Reservoir Dogs Store: The game gave me the achievement for securing all loot with 1 bag unaccounted for, according to WolfHUD. We searched high and low and could not locate another lootable item. Additionally, it seems like loot in the back of the store and in the vault counts before players can access it, which is not the case on similar heists like Hotline Miami Day 1.

Breakin' Feds: The potential loot counter for the heist is always +1 to the number of bags available for pickup. The Kotaru box doesn't count until Garrett's safe is open, but the evidence room loot does. Except the evidence room doesn't hold enough bags to account for the loot display.

Henry's Rock: The loot counter shows 5 potential loot items when only one exists - the Artefact in Archeology. I don't know what the other 4 are, but I suspect they may be the painting props in Archeology. Additionally, the number of openable crates listed seem +1 to the number of actual openable crates on the map. We've searched everywhere and weren't able to find that last crate to open.

## Steps to Reproduce

1. Play any of the abovementioned maps.
2. Look at the loot counter.
3. Try to find all of the displayed loot/crates, which you can't because the display is incorrect.

## Context

* Operating System: Wintos 7 Professional x64
* BLT version: 2
* Other mods installed:

AFSF2
Announcer
Arabic Numerals for Infamy
Bag Contour
Bot Weapons
Civilian Marking in Stealth
DelayedCallsFix
Enhanced Hitmarkers
Fading Contour
Full Speed Swarm
Goonmod's Custom Waypoints
Iter
Joker Outlines
Keepers
Less Inaccurate Weapon Laser
Live Long and Prosper
Look My Gun
Marking
Monkeepers
More Weapon Stats
Moveable Intimidated Cop
Mr. Dr. Fantastic!
No Confirm Dialogs
No Duplicated Bullets
No Flashlight Glow
No Sway
playerdrop
Please, Go There
Reduced iframe damage
Static Recoil
Wac
WolfHUD
Yet Another Flashlight

## BLT log

The bug is constant, logs are inapplicable.

## Crash log(s)

Not a crash

**Comment by RealDovahbear** [comment]
I realize this is a little old, but this is not a bug. It's an easter egg only visible via a map editor or using some form of free cam or noclip. I asked about it on the steam forums, [peep this.](https://steamcommunity.com/app/218620/discussions/8/1698294337764170168/)

**Comment by Kamikaze94** [comment]
Forgot to tag any issues, unobtainable loot should be fixed with commit bb4e6b11fcdb349f400b13158495ba4fa93b45f5.

If the loot is being counted from the start or as you can access it, depends on how Overkill decides to set the triggers. Loot gets counted when it exists on the map and the interaction is active, so when you could grab it. Sometimes the intzeraction is available from the start (Diamond Store), sometimes it gets activated by some trigger (Jewelery Store).

I couldn't get the Breakin feds job to spawn more loot that it shoewed me... Could be bad luck tough...

On Henry's Rock, you were correct about the bags :)
On my run, I could find all of the openable crates. Not sure if they are randomized, and if there is a chance for one out of reach though...

**Comment by Malidictus** [comment]
Thank you kindly for having a look :) I tried out the heists I remember seeing non-existent loot on. Reservoir Dogs and Breakin' Feds both worked out fine - the amount of loot listed in WolfHUD's loot display exactly matched the amount of loot I could find.

Henry's Rock is still a bit weird, though. You were completely correct - there are no non-existent crates. I'd failed to find one of them. However, there's still a single unobtainable bag, at least sometimes. It seems like the Archaeology artefact always shows up as a bag, whether you get the Archaeology section or not. I didn't on my run (got Biolabs and Archives), so I ended up with 1 unbagged item that I couldn't access.

Am I still just missing something on Henry's Rock, or is the 1 bag really the Archaeology artefact? If so, is a fix even possible? Either way, thank you for the fix :)
